MobileWallet/TariLib/Tor/OnionManager.swift

```diff
@@ -0,0 +1,310 @@
+/*
+ * Onion Browser
+ * Copyright (c) 2012-2018, Tigas Ventures, LLC (Mike Tigas)
+ *
+ * This file is part of Onion Browser. See LICENSE file for redistribution terms.
+ */
+// swiftlint:disable all
+import Foundation
+import Reachability
+import Tor
+
+protocol OnionManagerDelegate {
+    func torConnProgress(_: Int)
+    func torConnFinished(configuration: URLSessionConfiguration)
+    func torConnError()
+}
+
+public class OnionManager: NSObject {
+    public enum TorState: Int {
+        case none
+        case started
+        case connected
+        case stopped
+    }
+
+    public static let shared = OnionManager()
+    
+    public static var cookie: Data = {
+        let cookieURL = OnionManager.torBaseConf.dataDirectory!.appendingPathComponent("control_auth_cookie")
+        let cookie = try! Data(contentsOf: cookieURL)
```

Comment:
we should avoid also too. use try or try? https://medium.com/@JoyceMatos/error-handling-in-swift-3-try-try-and-try-f19705e32ff4